util/flocks/src/tfrwlock/mod.rs

```diff
@@ -5,52 +5,210 @@ pub use write_guard::TfrwLockWriteGuard;
 pub use read_guard::TfrwLockReadGuard;
 
 use tokio::sync::RwLock;
-use crate::frwlock::{FrwLock, FrwLockError};
-use rustix::fd::AsFd;
 use thiserror::Error;
-
+use rustix::{
+    fs::FlockOperation,
+    fd::AsFd,
+};
+use crate::asynchronous::flock;
+use crate::asynchronous::AsyncFlockError;
 
 #[derive(Debug, Error)]
 pub enum TfrwLockError {
```

Comment:
I don't like: 1) prefixes used where module paths can disambiguate instead; 2) cryptic prefixes.

Since the flocks API user is unlikely to mix the two kinds of locks in the same piece of code, I think it's better to give them the same name: `FileRwLock`, likewise for the ancillary guard and error types. And distinguish the two kinds by module paths: `sync::FileRwLock` for the sync version, `tokio::FileRwLock` for the async-friendly variant (to highlight that it requires the tokio runtime).